crates/resolver/src/name_server_pool.rs

```diff
@@ -472,6 +473,11 @@ impl PoolContext {
 pub struct NameServerTransportState(HashMap<IpAddr, ProtocolTransportState>);
 
 impl NameServerTransportState {
+    /// Return the count of nameservers with protocol transport state.
+    pub fn nameserver_count(&self) -> usize {
```

Comment:
Nit: to the bottom.

Reply:
This is `pub` and the other fns are `pub(crate)` so I think it's correct to leave here.